enterprise/cli/provisionerdaemonstart_test.go

```diff
@@ -49,7 +49,7 @@ func TestProvisionerDaemon_PSK(t *testing.T) {
 		pty.ExpectNoMatchBefore(ctx, "check entitlement", "starting provisioner daemon")
 		pty.ExpectMatchContext(ctx, "matt-daemon")
 
-		var daemons []codersdk.ProvisionerDaemon
+		var daemons []codersdk.ProvisionerDaemonWithStatus
```

Comment:
Hm.. is there an option not to change this entity and keep `codersdk.ProvisionerDaemon`? Perhaps we could simply add new optional properties.

Reply:
It felt weird to leave fields that are occasionally populated and other times not (especially the status field). I don't have a strong opinion here though, I just saw other similar use-cases in the code-base and decided to keep it separate.

I can move this all under `codersdk.ProvisionerDaemon` if you think it's better.

Reply:
I'll defer to others but `WithStatus` seems a bit awkward.

Reply:
It seems strange to sometimes have a provisioner daemon sometimes with status and sometimes without. I'd be more in favour of always return the provisioner daemon status, but I get the feeling that making that change would inflate the size of this PR significantly.

Reply:
@johnstcn yeah, it's a bit unfortunate but since getting the status requires querying the jobs table, I didn't want to make that change across the board in all places were querying the daemons and don't care about the status.

If we merge the structures, we could perhaps have the status field nullable/pointer to avoid any confusion why it's present or not.